<launch> 

	<!-- Tests the wobbler stack when used with physical hardware (notably, the roboteq based odometry frames and the Hokuyo and dynamixel data sources) -->

	<!-- Start all the nodes that are to be under test -->
	<include file="$(find babs_lidar_wobbler)/launch/start_wobbler.launch" /> 

	<!-- Start all the nodes that support simulation of the system -->
	<include file="$(find babs_nav)/launch/init_hw.launch"/> 

	<!-- Start node that will test the wobbler nodes -->
	<node pkg="babs_wobbler_tests" type="wobbler_integration_test" name="wobbler_integration_test" args="" output="screen" > 

		<!-- Test params to test the test --> 

		<param name="front_hokuyo_scan_name" value="/front_wobbler/scan" />
		<param name="front_transformed_scan_cloud_name" value="/front_wobbler/scan_cloud" />
		<param name="front_stitched_cloud_name" value="/front_wobbler/point_cloud" />
		<param name="front_motor_angle_name" value="/front_wobbler/cmd" />
		<param name="front_motor_command_name" value="/front_wobbler/cmd" />

		<param name="rear_hokuyo_scan_name" value="/rear_wobbler/scan" />
		<param name="rear_transformed_scan_cloud_name" value="/rear_wobbler/scan_cloud" />
		<param name="rear_stitched_cloud_name" value="/rear_wobbler/point_cloud" />
		<param name="rear_motor_angle_name" value="/rear_wobbler/cmd" />
		<param name="rear_motor_command_name" value="/rear_wobbler/cmd" />

		<param name="start_testing" value="1.0" />

	</node>

	<!-- NOTE: These topics are blank if not added as above, because Hokuyo nodes will fail and not post topics if there is no hardware to attach to. -->

	<!-- TEMPLATE
	<node pkg=" "  type=" " name=" " output=" " > 
	</node> -->

</launch>
